.github/workflows/gen_gbs_result_badge.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+##
+# SPDX-License-Identifier: LGPL-2.1-only
+# Copyright (C) 2024 Samsung Electronics
+#
+# @file gen_gbs_result_badge.py
+# @brief A tool for generating a github badge image in svg format
+# @author Gichan Jang <gichan2.jang@samsung.com>
+#
+import sys
+import os
+import json
+from pybadges import badge
+
+##
+# @brief Save the badge on the given path
+# @param[in] path A file path to save the svg file
+# @param[in] badge github badge svg file
+def save_badge (path, badge):
+    file = open(path, "w")
+    file.write(badge)
+    file.close()
+
+##
+# @brief Generate a github badge svg file representing daily build result
+# @param[in] html build result of arch.
+# @param[in] path A directory path to save the svg file
+def gen_build_result_badge(json_str, out_dir):
+    daily_result="success"
+    for arch in json_str:
+        result = json_str[arch]
+        daily_color = color= 'green'
+        if result == 'failure':
+          color = 'red'
+          daily_result = result
+          daily_color = 'red'
+
+        s = badge(left_text='test', right_text=result, right_color=color)
+        path = os.path.join(out_dir, arch + '_reuslt.svg')
+        save_badge (path, s)
```

Comment:
If we convert a JSON string into a dictionary using the ```json.load()``` method and try to access the key called ```[arch]``` in the dictionary, a ```KeyError``` may occur if the key does not exist.
Although it is not strictly necessary, it is recommended to handle ```KeyError``` in case of such exceptions.
How about you?
```
try:
        result = json_str[arch]
        daily_color = color= 'green'
        if result == 'failure':
          color = 'red'
          daily_result = result
          daily_color = 'red'

        s = badge(left_text='test', right_text=result, right_color=color)
        path = os.path.join(out_dir, arch + '_reuslt.svg')
        save_badge (path, s)
exception KeyError as e:
         print(f"KeyError occurred while processing {arch}: {str(e)}")    
         continue
```